py/client-ticking/build.gradle

```diff
@@ -0,0 +1,100 @@
+    networkName.set "pydeephaven-network-${randomSuffix}"
+}
+
+def buildPyClientTicking = Docker.registerDockerTask(project, 'pyClientTicking') {
+    // Only tested on x86-64, and we only build dependencies for x86-64
+    platform = 'linux/amd64'
+
+    copyIn {
+        from(layout.projectDirectory) {
+            include 'setup.py'
+            include 'README.md'
+            include 'src/**'
+        }
+    }
+    copyOut {
+        into layout.buildDirectory.dir('wheel')
+    }
+    dockerfile {
+        from('deephaven/py-client:local-build')
+        runCommand("""mkdir -p \\
+                        /out \\
+                        ${prefix}/log \\
+                        ${prefix}/src/py-client-ticking/src
+                   """)
+        copyFile('setup.py', "${prefix}/src/py-client-ticking")
+        copyFile('README.md', "${prefix}/src/py-client-ticking")
+        copyFile('src/', "${prefix}/src/py-client-ticking/src/")
+        runCommand("PREFIX=${prefix}; " +
+                   '''set -eux ; \
+                     cd "${PREFIX}/src/py-client-ticking"; \
+                     . "${PREFIX}/env.sh"; \
+                     MAKEFLAGS="-j${NCPUS}" \
+                       CFLAGS="-I${DHCPP}/include" \
+                       LDFLAGS="-L${DHCPP}/lib" \
+                       python3 setup.py build_ext -i; \
+                     python3 setup.py bdist_wheel; \
+                     pip3 install --force --no-deps dist/*.whl; \
+                     ln dist/*.whl /out; \
+                     cd /; \
+                     rm -fr "${PREFIX}/src/py-client-ticking"
+                   ''')
+    }
+    parentContainers = [ project.tasks.getByPath(':py-client:pyClient') ]
+}
+
+def testPyClientTicking = Docker.registerDockerTask(project, 'testPyClientTicking') {
+    // Only tested on x86-64, and we only build dependencies for x86-64
+    platform = 'linux/amd64'
+    copyIn {
+        from(layout.projectDirectory) {
+            include 'tests/**'
+        }
+    }
+    dockerfile {
+        from('deephaven/py-client-ticking:local-build')
+        runCommand("PREFIX=${prefix}; " +
+                   '''set -eux ; \
+                      pip3 install --force --no-deps /out/*.whl; \
+                      rm -fr /out; \
+                      mkdir -p \
+                        /out/report \
+                        /project/tests                      
+                   ''')
+        copyFile('tests/', "/project/tests/")
+        workingDir('/project')
+        //
+        // Setup for test run.
+        //
+        environmentVariable 'DH_HOST', deephavenDocker.containerName.get()
+        environmentVariable 'DH_PORT', '10000'
+    }
+    containerDependencies.dependsOn = [deephavenDocker.healthyTask]
+    containerDependencies.finalizedBy = deephavenDocker.endTask
+    network = deephavenDocker.networkName.get()
+    parentContainers = [ project.tasks.getByName('pyClientTicking') ]
```

Comment:
```suggestion
    parentContainers = [ buildPyClientTicking ]
```

Reply:
We discussed over slack, doing that results in gradle failure:
```
cfs@caicai 16:33:34 ~/dh/oss1/deephaven-core
$ git diff
diff --git a/py/client-ticking/build.gradle b/py/client-ticking/build.gradle
index 650973a94c..95584ddbe6 100644
--- a/py/client-ticking/build.gradle
+++ b/py/client-ticking/build.gradle
@@ -89,7 +89,7 @@ def testPyClientTicking = Docker.registerDockerTask(project, 'testPyClientTickin
     containerDependencies.dependsOn = [deephavenDocker.healthyTask]
     containerDependencies.finalizedBy = deephavenDocker.endTask
     network = deephavenDocker.networkName.get()
-    parentContainers = [ project.tasks.getByName('pyClientTicking') ]
+    parentContainers = [ buildPyClientTicking ]
     entrypoint = ['python3', '-m', 'xmlrunner', 'discover', 'tests', '-v', '-o', '/out/report']
     copyOut {
         into layout.buildDirectory.dir('test-results')


```

```
cfs@caicai 16:33:36 ~/dh/oss1/deephaven-core
$ ./gradlew :py-client-ticking:testPyClientTicking

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':py-client-ticking'.
> Could not create task ':py-client-ticking:testPyClientTicking'.
   > Could not create task ':py-client-ticking:testPyClientTickingMakeImage'.
      > class org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider_Decorated cannot be cast to class org.gradle.api.Task (org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider_Decorated and org.gradle.api.Task are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @66cd51c3)
[...]
```